File: scalechain-proto-codec/src/test/scala/io/scalechain/blockchain/proto/codec/indexparts/OneByteCodecSpec.scala
package io.scalechain.blockchain.proto.codec.indexparts

import io.scalechain.blockchain.proto.OneByte
import io.scalechain.blockchain.proto.codec.{OneByteCodec, PayloadTestSuite}
import io.scalechain.util.HexUtil._

class OneByteCodecSpec extends PayloadTestSuite[OneByte]  {

  val codec = OneByteCodec.codec

  val payload = bytes(
    """
       61
    """)

  val message = OneByte('a')
}

File: scalechain-proto-codec/src/test/kotlin/io/scalechain/blockchain/proto/codec/indexparts/OneByteCodecSpec.kt
package io.scalechain.blockchain.proto.codec.indexparts

import io.kotlintest.KTestJUnitRunner
import io.scalechain.blockchain.proto.OneByte
import io.scalechain.blockchain.proto.codec.OneByteCodec
import io.scalechain.blockchain.proto.codec.PayloadTestSuite
import io.scalechain.util.HexUtil.bytes
import org.junit.runner.RunWith

@RunWith(KTestJUnitRunner::class)
class OneByteCodecSpec : PayloadTestSuite<OneByte>()  {

  override val codec = OneByteCodec

  override val payload = bytes(
    """
       61
    """)

  override val message = OneByte('a'.toByte())
}
